hamaddr/src/lib.rs

```diff
@@ -0,0 +1,338 @@
+use anyhow::{bail, format_err};
+use std::convert::TryInto;
+use std::fmt;
+
+type Result<T> = std::result::Result<T, anyhow::Error>;
+
+/// Eui48 represents an EUI48 MAC address.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+pub struct Eui48(pub [u8; 6]);
+
+impl Eui48 {
+    /// Creates a new Eui48 from the given six octets.
+    pub const fn new(addr: [u8; 6]) -> Eui48 {
+        Eui48(addr)
+    }
+}
+/// Converts an Eui48 into a HamAddr
+impl Into<HamAddr> for Eui48 {
+    fn into(self: Self) -> HamAddr {
+        let mut octets = self.0;
+        octets[0] &= 0b1111_1101;
+        octets.rotate_left(1);
+        let mut bytes = [0; 8];
+        bytes[..6].copy_from_slice(&octets);
+        let addr = u64::from_be_bytes(bytes);
+        HamAddr(addr)
+    }
+}
+
+/// Formats an Eui48 for display.
+impl fmt::Display for Eui48 {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let bs = &self.0;
+        write!(
+            f,
+            "{:02x}:{:02x}:{:02x}:{:02x}:{:02x}:{:02x}",
+            bs[0], bs[1], bs[2], bs[3], bs[4], bs[5]
+        )
+    }
+}
+
+/// Eui64 represents an EUI64 MAC address.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+pub struct Eui64(pub [u8; 8]);
+
+impl Eui64 {
+    /// Creates a new Eui64 from the given 8 octets.
+    pub const fn new(addr: [u8; 8]) -> Eui64 {
+        Eui64(addr)
+    }
+}
+
+/// Formats an Eui64 for display.
+impl fmt::Display for Eui64 {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let bs = &self.0;
+        write!(
+            f,
+            "{:02x}:{:02x}:{:02x}:{:02x}:{:02x}:{:02x}:{:02x}:{:02x}",
+            bs[0], bs[1], bs[2], bs[3], bs[4], bs[5], bs[6], bs[7]
+        )
+    }
+}
+
+/// Converts an Eui64 into a HamAddr.
+impl Into<HamAddr> for Eui64 {
```

Comment:
Same here for `TryInfo`

Reply:
Interface changed, with a TODO to add proper error checking.